### PR TITLE
Linux: Converts passthrough syscalls to direct passthrough handlers

### PR DESCRIPTION
**Pull request by Sonicadvance1**
Reimagining of #3355 without any json generators or new concepts.

Fixes some mislabeling of system calls. Some getting inlined when they shouldn't be, a lot not getting inlined when they can be.

This really cleans up the syscall implementation, all syscalls that can be passthrough implementations require a very small two line declaration.
Additionally cleans up a bit of implementation cruft where some passthrough syscalls were using the glibc syscall handler, and some were using the glibc implementation. We have had multiple issues in the past where the glibc implementation does something subtly different than the raw syscall and breaks things. Now all passthrough handlers do a system call directly, removing at least one indirection and some ambiguity.

This makes it significantly easier to add new passthrough syscalls as well. Only need to do a version check and add the three lines per syscall. Which there are new syscalls incoming that we will want to add.

Tangible improvements:
- Syscalls are lower overhead than ever.
- When I'm adding more syscalls I have less chance of mucking it up.

### CODE REVIEW

**Review by lioncash — APPROVED** [review]

**Comment by Sonicadvance1** [comment]
Tony is wanting to review this. So holding off on merging until they get to it.

**Comment by lioncash** [comment]
Yeah, I was going to leave this one so more than one person can sign off on it, since it's quite a large change

File: Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/NotImplemented.cpp [hunk 1]
```diff
@@ -30,6 +30,8 @@ namespace FEX::HLE {
   // these are removed/not implemented in the linux kernel we present
 
   void RegisterNotImplemented(FEX::HLE::SyscallHandler *Handler) {
+      REGISTER_SYSCALL_NOT_IMPL(ustat);
+      REGISTER_SYSCALL_NOT_IMPL(sysfs);
```
Comment (neobrain) [inline Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/NotImplemented.cpp:34]: Looks like these were changed from placeholder implementations to fully unimplemented ones. Probably better to split this type of change into a separate commit for reference. (If this causes unexpected problems, it's easier to pin down the problem if it's not hidden in a 2000 line diff :) )

**Review by neobrain — APPROVED** [review]
Haven't encountered any problems while running this locally for a bit. Really great job on the code savings!

File: Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/EPoll.cpp [hunk 2]
```diff
@@ -19,16 +19,10 @@ namespace FEX::HLE {
   void RegisterEpoll(FEX::HLE::SyscallHandler *Handler) {
     using namespace FEXCore::IR;
 
-    REGISTER_SYSCALL_IMPL_PASS_FLAGS(epoll_create, SyscallFlags::OPTIMIZETHROUGH | SyscallFlags::NOSYNCSTATEONENTRY,
```
[the remark this answers is not in the document]
Comment (Sonicadvance1) [reply Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/EPoll.cpp:22]: Removed.

File: Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/NotImplemented.cpp (hunk 1 above)
Comment (Sonicadvance1) [reply Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/NotImplemented.cpp:34]: Moved to a commit prior to the large one.

File: Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/Passthrough.cpp [hunk 3]
```diff
@@ -0,0 +1,889 @@
...
+
+#include <FEXCore/IR/IR.h>
+
+#include <stdint.h>
+#include <sys/epoll.h>
+
+namespace FEX::HLE {
+#ifdef _M_ARM_64
+template<int syscall_num>
+requires (syscall_num != -1)
```
[the remark this answers is not in the document]
Comment (Sonicadvance1) [reply Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/Passthrough.cpp:21]: Syscall enum values are declared in `Source/Tools/LinuxEmulation/LinuxSyscalls/{Arm64,x32,x64}/SyscallsEnum.h`.
Syscalls that don't exist on that platform are defined as -1.

This catches the programming error of trying to passthrough a syscall that doesn't exist on the host platform and will result in a compile error instead of unexpected ENOSYS.

This was very nice when converting all the passthrough operations and finding ones that were getting passed through when they shouldn't be. Additionally this will be useful in the future when adding new syscalls to ensure we don't accidentally add one that doesn't exist on AArch64. This isn't very frequent but in kernel 6.6 `map_shadow_stack` was added on x86-64 but doesn't exist on AArch64.

[the remark this answers is not in the document]
Comment (neobrain) [reply Source/Tools/LinuxEmulation/LinuxSyscalls/Syscalls/Passthrough.cpp:21]: Ah yes, makes sense. Thanks for the explanation!
